blocks/product-support/product-support.js

```diff
@@ -127,6 +137,82 @@ export default async function decorate(block) {
     }
   };
 
+  const videoAssets = getAssets(product, allDocuments, allDocuments)
+    .filter((asset) => parseYoutubeCode(asset.URL));
```

Comment:
the `parseYoutubeCode` function is called a total of 3 times for the each URL: once here, then two times in `createDomStructure`. You eliminate the redundant calls by storing the the parsed code in the asset:

```
const videoAssets = getAssets(product, allDocuments, allDocuments)
  .map((asset) => ({ ...asset, youtubeCode: parseYoutubeCode(asset.URL) }))
  .filter((asset) => asset.youtubeCode);
```

and then further down reference like this:
```
...
  children: [
    {
      type: 'a',
      attributes: { href: `https://www.youtu.be/${asset.youtubeCode}` },
    },
  ],
...
```